File: app/src/main/java/org/thoughtcrime/securesms/mms/SentMediaQuality.java
package org.thoughtcrime.securesms.mms;

import androidx.annotation.NonNull;

/**
 * Quality levels to send media at.
 */
public enum SentMediaQuality {
  STANDARD(0),
  HIGH(1);


  private final int code;

  SentMediaQuality(int code) {
    this.code = code;
  }

  public static @NonNull SentMediaQuality fromCode(int code) {
    if (HIGH.code == code) {
      return HIGH;
    }
    return STANDARD;
  }

  public int getCode() {
    return code;
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/mms/SentMediaQuality.kt
package org.thoughtcrime.securesms.mms

import android.content.Context
import androidx.annotation.StringRes
import org.thoughtcrime.securesms.R

/**
 * Quality levels to send media at.
 */
enum class SentMediaQuality(@JvmField val code: Int, @param:StringRes private val label: Int) {
  STANDARD(0, R.string.DataAndStorageSettingsFragment__standard),
  HIGH(1, R.string.DataAndStorageSettingsFragment__high);

  companion object {
    @JvmStatic
    fun fromCode(code: Int): SentMediaQuality {
      return if (HIGH.code == code) {
        HIGH
      } else {
        STANDARD
      }
    }

    fun getLabels(context: Context): Array<String> {
      return entries.map { context.getString(it.label) }.toTypedArray()
    }
  }
}
